### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by kamilwaz**
Updates Phoenix and other dependencies to latest versions. Alongside, Erlang was updated to 25.0 and Elixir to 1.13.

### CODE REVIEW

**Comment by kamilwaz** [comment]
Erlang cannot be upgraded to 25.x because `distillery` doesn't support this version yet. The change that blocks it: https://github.com/erlang/otp/commit/5ea1d07c17447ff114879b292d4b9d461c191ccd.

File: Dockerfile [hunk 1]
```diff
@@ -23,7 +30,7 @@ RUN mix do certs.dev, distillery.release
 RUN tar -czf mongoose_push.tar.gz -C _build/prod/rel/mongoose_push .
 
 
-FROM debian:stretch-slim
+FROM ubuntu:20.04
```
Comment (michalwski) [inline Dockerfile]: This is the final docker image inside which the built MongoosePush will run. We used `debian:stretch-slim` to make the final image as small as possible. Do we have something similar to `ubuntu:20.04` ?

File: test/unit/test_helper.exs [hunk 2]
```diff
@@ -1,2 +1,4 @@
+Code.put_compiler_option(:warnings_as_errors, true)
```
Comment (michalwski) [inline test/unit/test_helper.exs:1]: 👍

**Review by michalwski — COMMENTED** [review]
Looks good, I had some comments/questions.

File: Dockerfile (hunk 1 above)
Comment (kamilwaz) [reply Dockerfile]: Unfortunately, not. I changed the final image back to `debian`.

**Review by michalwski — APPROVED** [review]
Ok, approving it. Go ahead and merge.

**Comment by michalwski** [comment]
> Erlang cannot be upgraded to 25.x because distillery doesn't support this version yet. The change that blocks it: https://github.com/erlang/otp/commit/5ea1d07c17447ff114879b292d4b9d461c191ccd.

In fact, we could move away from `distillery` to the built-in release mechanism. The built-in got improved in Elixir after this project was actively developed so probably there's no reason to use the `distillery` anymore.

**Pull request merged**